### DISCUSSION TITLE
interpolation

### DISCUSSION THREAD

**Discussion by MarinCronograph** [gempy-project/gempy#939] [General]
Hello,

I am guessing that with this line of code we are imposing interpolation, I tried to find settings used here, f ex.what interpolation method and other settings connected to that.

Tried with print(geo_model.interpolation_options). How to get all the settings used here and how to change them to fit to my needs

rgds

**Comment by javoha** [comment]
Hi,
so this dives a little deeper into the core of gempy. The interpolation method is always a universal cokriging approach. If you want a better understanding of what is happening I would really suggest you look into the literature linked on the frontpage. 

I might be able to answer specific questions once you are through that.

Cheers
Jan

**Reply by MarinCronograph** [reply]
hi @javoha ,

i will take a look here about interpolation
https://www.gempy.org/tutorials

**Reply by javoha** [reply]
Thats always a good idea, but for a better understanding of the underlying concepts you might want to read the literature linked on the frontpage:

de la Varga, M., Schaaf, A., and Wellmann, F. (2019). [GemPy 1.0: open-source stochastic geological modeling and inversion](https://gmd.copernicus.org/articles/12/1/2019/gmd-12-1-2019.pdf), Geosci. Model Dev., 12, 1-32.
Wellmann, F., & Caumon, G. (2018). [3-D Structural geological models: Concepts, methods, and uncertainties.](https://hal.univ-lorraine.fr/hal-01921494/file/structural_models_for_geophysicsHAL.pdf) In Advances in Geophysics (Vol. 59, pp. 1-121). Elsevier.
Calcagno, P., Chilès, J. P., Courrioux, G., & Guillen, A. (2008). [Geological modelling from field data and geological knowledge: Part I. Modelling method coupling 3D potential-field interpolation and geological rules](https://www.sciencedirect.com/science/article/abs/pii/S0031920108001258). Physics of the Earth and Planetary Interiors, 171(1-4), 147-157.
Lajaunie, C., Courrioux, G., & Manuel, L. (1997). [Foliation fields and 3D cartography in geology: principles of a method based on potential interpolation](https://link.springer.com/article/10.1007/BF02775087). Mathematical Geology, 29(4), 571-584.

Cheers,
Jan

**Reply by MarinCronograph** [reply]
hello @javoha , i tried this to change to -gaussian-  but it didnt help

geo_model.interpolation_options.kernel_options.kernel_function = AvailableKernelFunctions.GAUSSIAN

do u have any advice  ?

rgds

Marin Mirosevic

**Reply by javoha** [reply]
Hi,
maybe you could specify a little bit what "it didnt help" means. What does your model look like and what would you like to achieve?

**Reply by MarinCronograph** [reply]
@javoha 

Sorry for not being clear .

I tried to change interpolation algorith from -cubic-to -Gaussian.

I have used this line of code:  geo_model.interpolation_options.kernel_options.kernel_function = AvailableKernelFunctions.GAUSSIAN

Seems that syntax might be wrong. I am using 2024.2.0.2 gempy version.

rgds,

MArin Mirosevic

**Reply by javoha** [reply]
Hi,
I have to check it out - but my question is more in the direction of what you want to achieve. Like what is your model not doing that you want it to do?

**Reply by MarinCronograph** [reply]
@javoha 

I am learning this and testing different things.

So i wanted to change algorithm

Rgds

Marin

**Comment by javoha** [comment]
Ok let's see. This is rather advanced and I would be cautious using it. As far as I can see we do not have a Gaussian covariance function implemented yet (feel free to do and create a pull request - that would be great) but we do have exponential and matern models as alternatives to the default cubic function that is supported by literature.
I did not find a simple setter function for this (maybe I have to take a closer look) but here is how it can be done:

```
import gempy_engine as gpe
geo_data.interpolation_options.kernel_options.kernel_function = gpe.core.data.kernel_classes.kernel_functions.AvailableKernelFunctions.exponential
```
I hope this helps!
